micromanager_gui/main_window.py

```diff
@@ -139,36 +141,82 @@ def __init__(self, viewer: napari.viewer.Viewer, remote=True):
         self.bit_comboBox.currentIndexChanged.connect(self.bit_changed)
         self.bin_comboBox.currentIndexChanged.connect(self.bin_changed)
 
+        @self.viewer.mouse_drag_callbacks.append
```

Comment:
another thing that would be desirable, would be to move as much of the sample-explorer specific logic, into the sample-explorer widget. In the interest of keeping things modular, it's best to limit cross-widget communication to the bare minimum.

It looks to me like this whole mouse callback is really just for the sample explorer, right? If so, it should be in the sample explorer file. If you need to pass the viewer to the sample-explorer constructor, that's fine